oshdb-api/src/main/java/org/heigit/bigspatialdata/oshdb/api/mapreducer/backend/MapReducerIgniteLocalPeek.java

```diff
@@ -226,34 +241,73 @@ public boolean isActive() {
       return this.notCanceled;
     }
 
-    public abstract S execute(Ignite node);
+    private class CellKeysIterator implements Iterator<Long> {
+      Iterator<Pair<CellId, CellId>> cellIdRanges;
+      ArrayList<Long> buffer;
+
+      // a buffer of about 1M interleaved cellIds
+      static final int BUFFER_SIZE = 1_000_000;
+
+      CellKeysIterator(Iterable<Pair<CellId, CellId>> cellIdRanges) {
+        this.cellIdRanges = cellIdRanges.iterator();
+        buffer = new ArrayList<>(BUFFER_SIZE);
+      }
+
+      private void fillBuffer() {
+        buffer.clear();
+        while (buffer.size() < BUFFER_SIZE) {
+          if (!cellIdRanges.hasNext()) {
+            break;
+          }
+          Pair<CellId, CellId> cellIdRange = cellIdRanges.next();
+          int level = cellIdRange.getLeft().getZoomLevel();
+          long fromId = cellIdRange.getLeft().getId();
+          long toId = cellIdRange.getRight().getId();
+          for (long id = fromId; id < toId; id++) {
+            buffer.add(new CellId(level, id).getLevelId());
+          }
```

Comment:
we could put this code 
```java
          Pair<CellId, CellId> cellIdRange = cellIdRanges.next();
          int level = cellIdRange.getLeft().getZoomLevel();
          long fromId = cellIdRange.getLeft().getId();
          long toId = cellIdRange.getRight().getId();
          for (long id = fromId; id < toId; id++) {
            buffer.add(new CellId(level, id).getLevelId());
          }
```
in a flatMap step of the cellIdRanges stream. that makes pulling out BUFFER_SIZE ids more predictable.
At the moment it could happen that we "overfill" the buffer with more than BUFFER_SIZE ids.

Reply:
does https://github.com/GIScience/oshdb/pull/64/commits/0f50172a43a14a7f48c166934c1bb44441c150cf#diff-2f3781c256730a1206a3368a569b94f6R255 look ok?